client/src/app/services/authentication.service.ts

```diff
@@ -18,19 +19,25 @@ export class AuthenticationService {
     return this.currentUserSubject.value;
   }
 
-  public login(username: string, password: string):boolean{
-    const hardCodedUser:User = {
-      id:1,
-      userName:'admin',
-      password:'admin123',
-      token:'token'
+  public login(username: string, password: string): Observable<any> {
+    const user: User = {
+      id: 1,                     //do I need that?
+      userName: username,
+      password: password,
+      token: ''
     }
-    if(username === hardCodedUser.userName && password === hardCodedUser.password){
-        localStorage.setItem('currentUser', JSON.stringify(hardCodedUser));
-        this.currentUserSubject.next(hardCodedUser);
-        return true;
-    };
-    return false;
+    return this.apiService.login({ name: username, password: password }).pipe(
+      map((res: any) => {
+        user.token = res.result.data.token;
```

Comment:
if (res && res.token) {
                    localStorage.setItem('currentUser', JSON.stringify(res));
                    this.currentUserSubject.next(res);
                }

Reply:
In my res there's just a token, so should I change my UserModel to be 
```
export interface UserModel {
    token: string;
}
``` ?